opm/simulators/wells/WellGroupHelpers.hpp

```diff
@@ -29,15 +29,17 @@ namespace Opm {
     namespace wellGroupHelpers
     {
 
-    inline void setGroupControl(const Group& group, const Schedule& schedule, const int reportStepIdx, const bool injector, WellStateFullyImplicitBlackoil& wellState, std::ostringstream& ss) {
+    inline void setGroupControl(const Group& group, const Schedule& schedule, const Phase& topPhase, const int reportStepIdx, const bool injector, WellStateFullyImplicitBlackoil& wellState, std::ostringstream& ss) {
 
         for (const std::string& groupName : group.groups()) {
             const Group& groupTmp = schedule.getGroup(groupName, reportStepIdx);
-            setGroupControl(groupTmp, schedule, reportStepIdx, injector, wellState, ss);
-            if (injector)
-                wellState.setCurrentInjectionGroupControl(groupName, Group::InjectionCMode::FLD);
-            else
+            setGroupControl(groupTmp, schedule, topPhase, reportStepIdx, injector, wellState, ss);
+            if (injector) {
+                if (groupTmp.injection_phase() == topPhase || wellState.currentInjectionGroupControl(groupName) == Group::InjectionCMode::NONE) // only switch sub groups with same face or NONE
```

Comment:
Why do we switch NONE groups?

And: what happens if, in a chain of groups (A parent of B parent of C), if A switches to (say) oil rate control, B is still on gas control, but C is on oil control? Is that possible?

Also a small typo: face -> phase.

Reply:
Hierarchy: That should be fine. Keep in mind these are injectors and injecting oil is maybe not the best example. Say group A switch to water injector control, B keeps injecting gas, then C should change to FLD since it may need to be constraint to honor group As target.  

NONE: If group A with gas rate control is parent of a group B without any control. Group B will default to water phase. But it should be counted in.